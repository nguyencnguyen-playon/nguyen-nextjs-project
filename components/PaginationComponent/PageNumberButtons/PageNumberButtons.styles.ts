import { css } from 'styled-components';

export const styles = {
  pageNumberButtonsContainer: css<{ margin: string }>`
    display: flex;
    flex-direction: row;
  `,
  pageNumberButton: css<{ active: boolean; btnBackgroundColor: string }>`
    align-items: center;
    justify-content: center;
    width: 30px;
    height: 40px;
    border: 1px solid #dbdbdb;
    background-color: ${({ active, btnBackgroundColor }) => (active ? btnBackgroundColor : 'white')};
  `,
  pageNumberText: css<{ active: boolean; color: string }>`
    font-weight: 700;
    color: ${({ active, color }) => (active ? color : '#000')};
  `,
  navigatePageButton: css`
    align-items: center;
    justify-content: center;
    width: 40px;
    height: 40px;
    border: 1px solid #dbdbdb;
    background-color: white;
  `,
  navigatePageText: css<{ disabled: boolean }>`
    font-weight: 700;
    color: ${({ disabled }) => (disabled ? '#dbdbdb' : '#000')};
  `
};
