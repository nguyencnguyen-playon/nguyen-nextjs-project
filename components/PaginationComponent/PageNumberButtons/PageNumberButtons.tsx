import styled from 'styled-components';
import React from 'react';
import { styles } from './PageNumberButtons.styles';

const PageNumberButtonsContainer = styled.div<{ margin: string }>`
  ${styles.pageNumberButtonsContainer}
`;

const PageNumberButton = styled.button<{ active: boolean; btnBackgroundColor: string }>`
  ${styles.pageNumberButton}
`;

const PageNumberText = styled.div<{ active: boolean; color: string }>`
  ${styles.pageNumberText}
`;

const NavigatePageButton = styled.button`
  ${styles.navigatePageButton}
`;

const NavigatePageText = styled.div<{ disabled: boolean }>`
  ${styles.navigatePageText}
`;

type PageNumberButtonsProps = {
  margin: string;
  currentPage: number;
  handlePageChange: (newPage: number) => void;
  endPage: number;
  startPage: number;
  btnBackgroundColor: string;
  textBtnColor: string;
  totalPages: number;
};

export const PageNumberButtons = ({
  margin,
  currentPage,
  handlePageChange,
  endPage,
  startPage,
  btnBackgroundColor,
  textBtnColor,
  totalPages
}: PageNumberButtonsProps) => {
  if (endPage === startPage) {
    return;
  }
  return (
    <PageNumberButtonsContainer margin={margin}>
      <NavigatePageButton disabled={currentPage === 1} onClick={() => handlePageChange(currentPage - 1)}>
        <NavigatePageText disabled={currentPage === 1}>
          {'<'}
        </NavigatePageText>
      </NavigatePageButton>

      {[...Array(endPage - startPage + 1)].map((_, index) => {
        const pageNumber = startPage + index;
        return (
          <PageNumberButton
            key={`page-${pageNumber}`}
            onClick={() => handlePageChange(pageNumber)}
            active={pageNumber === currentPage}
            btnBackgroundColor={btnBackgroundColor}
          >
            <PageNumberText color={textBtnColor} active={pageNumber === currentPage}>
              {pageNumber}
            </PageNumberText>
          </PageNumberButton>
        );
      })}
      <NavigatePageButton
        disabled={currentPage === totalPages}
        onClick={() => handlePageChange(currentPage < totalPages ? currentPage + 1 : currentPage)}
      >
        <NavigatePageText disabled={currentPage === totalPages}>
          {'>'}
        </NavigatePageText>
      </NavigatePageButton>
    </PageNumberButtonsContainer>
  );
};
