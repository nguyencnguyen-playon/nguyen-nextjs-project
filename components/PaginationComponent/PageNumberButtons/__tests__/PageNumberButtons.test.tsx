import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { PageNumberButtons } from '../PageNumberButtons';

describe('PageNumberButtons', () => {
    const handlePageChangeMock = jest.fn();

    const defaultProps = {
        margin: '10px',
        currentPage: 1,
        handlePageChange: handlePageChangeMock,
        endPage: 10,
        startPage: 1,
        btnBackgroundColor: 'blue',
        textBtnColor: 'white',
        totalPages: 10,
    };

    it('should match snapshot', () => {
        const component = render(<PageNumberButtons {...defaultProps} />);
        expect(component).toMatchSnapshot();
    })

    it('renders the page number buttons correctly', () => {
        const { getByText } = render(<PageNumberButtons {...defaultProps} />);

        for (let i = defaultProps.startPage; i <= defaultProps.endPage; i++) {
            const pageNumberButton = getByText(i.toString());
            expect(pageNumberButton).toBeTruthy();
        }
    });

    it('calls handlePageChange when a page number button is clicked', () => {
        const { getByText } = render(<PageNumberButtons {...defaultProps} />);

        const pageNumberButton = getByText('2');
        fireEvent.click(pageNumberButton);

        expect(handlePageChangeMock).toHaveBeenCalledWith(2);
    });
});
