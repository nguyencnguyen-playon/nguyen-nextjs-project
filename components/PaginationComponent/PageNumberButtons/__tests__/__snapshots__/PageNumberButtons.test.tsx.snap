// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNumberButtons should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-beqWNU fdMZMg"
        margin="10px"
      >
        <button
          class="sc-hLseSp bnIruC"
          disabled=""
        >
          <div
            class="sc-eDDOjk jsLcAC"
            disabled=""
          >
            &lt;
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH kWcWOh"
        >
          <div
            class="sc-dmqGRC tZQpx"
            color="white"
          >
            1
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            2
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            3
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            4
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            5
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            6
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            7
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            8
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            9
          </div>
        </button>
        <button
          btnbackgroundcolor="blue"
          class="sc-gueXAH tDkzQ"
        >
          <div
            class="sc-dmqGRC emngKN"
            color="white"
          >
            10
          </div>
        </button>
        <button
          class="sc-hLseSp bnIruC"
        >
          <div
            class="sc-eDDOjk dwbRyI"
          >
            &gt;
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-beqWNU fdMZMg"
      margin="10px"
    >
      <button
        class="sc-hLseSp bnIruC"
        disabled=""
      >
        <div
          class="sc-eDDOjk jsLcAC"
          disabled=""
        >
          &lt;
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH kWcWOh"
      >
        <div
          class="sc-dmqGRC tZQpx"
          color="white"
        >
          1
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          2
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          3
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          4
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          5
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          6
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          7
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          8
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          9
        </div>
      </button>
      <button
        btnbackgroundcolor="blue"
        class="sc-gueXAH tDkzQ"
      >
        <div
          class="sc-dmqGRC emngKN"
          color="white"
        >
          10
        </div>
      </button>
      <button
        class="sc-hLseSp bnIruC"
      >
        <div
          class="sc-eDDOjk dwbRyI"
        >
          &gt;
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
